candidate_bestPractice_OrangeHRM/cypress/support/utils/apiUtils.ts

```diff
@@ -23,9 +26,92 @@ Cypress.Commands.add(
         failOnStatusCode: false,
       })
       .then((response) => {
-        expect(response.status).eq(200)
+        expect(response.status).eq(200);
       })
       .its("body");
+  }
+);
+declare global {
+  namespace Cypress {
+    interface Chainable {
+      employee: (
+        method: string,
+        requestUrl: string,
+        employeePayload: string
+      ) => Chainable<employeeResponse>;
+    }
+  }
+}
+
+Cypress.Commands.add(
+  "employee",
+  (method: string, requestUrl: string, employeePayload: string) => {
+    return cy
+      .api({
+        method: method,
+        url: requestUrl,
+        body: employeePayload,
+        failOnStatusCode: false,
+      })
+      .then((response) => {
+        expect(response.status).eq(200);
+      })
+      .its("body");
+  }
+);
+declare global {
+  namespace Cypress {
+    interface Chainable {
+      user: (
+        method: string,
+        requestUrl: string,
+        userPayload: string
+      ) => Chainable<userResponse>;
+    }
+  }
+}
 
+Cypress.Commands.add(
+  "user",
+  (method: string, requestUrl: string, userPayload: string) => {
+    return cy
+      .api({
+        method: method,
+        url: requestUrl,
+        body: userPayload,
+        failOnStatusCode: false,
+      })
+      .then((response) => {
+        expect(response.status).eq(200);
+      })
+      .its("body");
+  }
+);
+declare global {
+  namespace Cypress {
+    interface Chainable {
+      vacancy: (
+        method: string,
+        requestUrl: string,
+        vacancyPayload: string
+      ) => Chainable<vacancyResponse>;
+    }
+  }
+}
+
+Cypress.Commands.add(
+  "vacancy",
+  (method: string, requestUrl: string, vacancyPayload: string) => {
+    return cy
+      .api({
+        method: method,
+        url: requestUrl,
+        body: vacancyPayload,
+        failOnStatusCode: false,
+      })
+      .then((response) => {
+        expect(response.status).eq(200);
+      })
+      .its("body");
```

Comment:
No need to make the  add vacancy command

Reply:
I think I need to add a vacancy in order for me to be able to add a candidate to that specific vacancy to not end up not finding any vacancy to add a candidate to.